src/main/kotlin/org/malachite/estella/services/TaskStageService.kt

```diff
@@ -106,7 +108,7 @@ class TaskStageService(
 
 
     fun addResult(resultToAdd: TaskService.ResultToAdd) {
-        val resultToSave = taskResultRepository.findAll().firstOrNull { it.task.id == resultToAdd.task.id && it.taskStage.id == resultToAdd.taskStage.id }.let {
+        val resultToSave = taskResultRepository.findAll().firstOrNull { it.task.id == resultToAdd.task.id && it.taskStage.id == resultToAdd.taskStage.id && it.startTime != null }.let {
```

Comment:
Po pierwsze czemu nie używamy find'a, po drugie jeśli jest tyle warunków to można to przenieść do funkcji pomocniczej.